cdap-docs/build.sh

```diff
@@ -182,24 +188,43 @@ function build_javadocs() {
     local debug_flag=''
   fi
   if [[ ${javadoc_type} == ${ALL} ]]; then
-    javadoc_run="javadoc:aggregate"
+    javadoc_run="javadoc:aggregate -P release"
   else
-    javadoc_run="site"
+    javadoc_run="site -P templates"
   fi
   local start=`date`
   cd ${PROJECT_PATH}
   MAVEN_OPTS="-Xmx4g -XX:MaxPermSize=256m" # match other CDAP builds
-  mvn clean package ${javadoc_run} -P examples,templates,release -DskipTests -Dgpg.skip=true -DisOffline=false ${debug_flag}
-  warnings=$?
-  if [[ ${warnings} -eq 0 ]]; then
-    echo
-    echo "Javadocs Build Start: ${start}"
-    echo "                 End: `date`"
+  local temp_repo="-Dmaven.repo.local=/tmp/repo-cdap-docs/${GIT_BRANCH}"
+  echo "Using temp_repo '${temp_repo}'"
+  if [[ -d ${temp_repo} ]]; then
+    rm -rf ${temp_repo}
+  fi 
+  echo "========================================================"
+  echo "Building and installing CDAP to ${temp_repo}"
+  echo "command:"
+  echo "mvn clean install ${temp_repo} -P examples,templates,release -DskipTests -Dgpg.skip=true"
+  mvn clean install ${temp_repo} -P examples,templates,release -DskipTests -Dgpg.skip=true
```

Comment:
So you're duplicating an entire CDAP build, including downloading *ALL* dependencies, with no cache, on every build? This is excessive and increases bandwidth usage dramatically. Our dependency JARs should already be cached in our local Maven repository. We should use that, instead.

```
mkdir -p ${temp_repo}
if [[ -d ${HOME}/.m2/repository ]]; then
  cp -a ${HOME}/.m2/repository ${temp_repo}
  # Cleanup any CDAP JARs, so we use our own
  rm -rf ${temp_repo}/co/cask/cdap
fi
```